src/Approvals/SFA.DAS.Approvals/Application/Apprentices/Queries/GetReviewApprenticeshipUpdates/GetReviewApprenticeshipUpdatesQueryHandler.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.Approvals.Enums;
+using SFA.DAS.Approvals.Extensions;
+using SFA.DAS.Approvals.InnerApi;
+using SFA.DAS.Approvals.InnerApi.CommitmentsV2Api.Requests;
+using SFA.DAS.Approvals.InnerApi.CommitmentsV2Api.Responses;
+using SFA.DAS.Approvals.Services;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Extensions;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.Approvals.Application.Apprentices.Queries.GetReviewApprenticeshipUpdates
+{
+    public class GetReviewApprenticeshipUpdatesQueryHandler : IRequestHandler<GetReviewApprenticeshipUpdatesQuery, GetReviewApprenticeshipUpdatesQueryResult>
+    {
+        private readonly ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> _apiClient;
+        private readonly IProviderStandardsService _providerStandardsService;
+        private readonly ServiceParameters _serviceParameters;
+
+        public GetReviewApprenticeshipUpdatesQueryHandler(
+            ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> apiClient,
+            IProviderStandardsService providerStandardsService,
+            ServiceParameters serviceParameters)
+        {
+            _apiClient = apiClient;
+            _providerStandardsService = providerStandardsService;
+            _serviceParameters = serviceParameters;
+        }
+
+        public async Task<GetReviewApprenticeshipUpdatesQueryResult> Handle(GetReviewApprenticeshipUpdatesQuery request, CancellationToken cancellationToken)
+        {
+            var apprenticeshipTask = _apiClient.GetWithResponseCode<GetApprenticeshipResponse>(new GetApprenticeshipRequest(request.ApprenticeshipId));
+
+            var updatesTask = _apiClient.GetWithResponseCode<GetApprenticeshipUpdatesResponse>(new GetApprenticeshipUpdatesRequest(request.ApprenticeshipId, Convert.ToByte(ApprenticeshipStatus.WaitingToStart)));
+
+            await Task.WhenAll(updatesTask, apprenticeshipTask);
+
+            var apprenticeshipResponse = apprenticeshipTask.Result;
+            var apprenticeshipUpdateResponse = updatesTask.Result;
+
+            if (apprenticeshipUpdateResponse.StatusCode == HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+
+            apprenticeshipUpdateResponse.EnsureSuccessStatusCode();
+
+            var apprenticeshipUpdates = apprenticeshipUpdateResponse.Body;
+
+            if (apprenticeshipResponse.StatusCode == HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+
+            apprenticeshipResponse.EnsureSuccessStatusCode();
+
+            var apprenticeship = apprenticeshipResponse.Body;
+
+
+            if (!apprenticeship.CheckParty(_serviceParameters))
+            {
+                return null;
+            }
+
+            if (apprenticeshipUpdates.ApprenticeshipUpdates.Count == 1)
```

Comment:
I think we can check for > 1 and throw an exception here, so that the entire nested block is not longer nested.